web/portal/brand/src/entities/Carrier/Action/BalanceOperations.tsx

```diff
@@ -0,0 +1,198 @@
+import { EntityValues } from '@irontec/ivoz-ui';
+import { MoreMenuItem } from '@irontec/ivoz-ui/components/List/Content/Shared/MoreChildEntityLinks';
+import { StyledTableRowCustomCta } from '@irontec/ivoz-ui/components/List/Content/Table/ContentTable.styles';
+import {
+  OutlinedButton,
+  SolidButton,
+} from '@irontec/ivoz-ui/components/shared/Button/Button.styles';
+import {
+  ActionFunctionComponent,
+  ActionItemProps,
+} from '@irontec/ivoz-ui/router/routeMapParser';
+import { StyledDropdown } from '@irontec/ivoz-ui/services/form/Field/Dropdown/Dropdown.styles';
+import { StyledTextField } from '@irontec/ivoz-ui/services/form/Field/TextField/TextField.styles';
+import _ from '@irontec/ivoz-ui/services/translations/translate';
+import ReplayIcon from '@mui/icons-material/Replay';
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  SelectChangeEvent,
+  Tooltip,
+} from '@mui/material';
+import axios from 'axios';
+import { CurrencyPropertyList } from 'entities/Currency/CurrencyProperties';
+import { useEffect, useState } from 'react';
+import { useStoreActions } from 'store';
+
+import Carrier from '../Carrier';
+import { CarrierPropertyList } from '../CarrierProperties';
+
+const amountChoices = [
+  { id: '+', label: '+', operation: 'increment' },
+  { id: '-', label: '-', operation: 'decrement' },
+];
+
+const BalanceOperations: ActionFunctionComponent = (props: ActionItemProps) => {
+  const { row, variant = 'icon' } = props;
+  const [amountChoice, setAmountChoice] = useState(amountChoices[0].id);
+  const [amountValue, setAmountValue] = useState('0.00');
+  const [open, setOpen] = useState(false);
+  const [currencySymbol, setCurrencySymbol] = useState('');
+  const apiPost = useStoreActions((actions) => {
+    return actions.api.post;
+  });
+  const cancelTokenSource = axios.CancelToken.source();
```

Comment:
use useCancelToken from `ivoz-ui/hooks/useCancelToken.tsx` better or just don't use a cancelToken at all. Whatever you prefer